File: client/src/types/Notification.ts
export interface Notification {
  notificationId: number;
  userId: number;
  title: string;
  message: string;
  type: NotificationType;
  priority: NotificationPriority;
  isRead: boolean;
  createdDate: Date;
  readDate?: Date;
  issueId?: number;
  orderId?: number;
  paymentId?: number;
}

export type NotificationType = 'issue' | 'payment' | 'order' | 'general';
export type NotificationPriority = 'low' | 'medium' | 'high' | 'urgent';

export interface CreateNotificationRequest {
  userId: number;
  title: string;
  message: string;
  type: NotificationType;
  priority: NotificationPriority;
  issueId?: number;
  orderId?: number;
  paymentId?: number;
}

export interface UpdateNotificationRequest {
  isRead?: boolean;
  readDate?: Date;
} 

File: server/Models/Notification.cs
using System.ComponentModel.DataAnnotations;

namespace server.Models
{
    public class Notification
    {
        [Key]
        public int NotificationId { get; set; }
        
        public int UserId { get; set; }
        public UserAccount User { get; set; }
        
        public string Title { get; set; } = string.Empty;
        public string Message { get; set; } = string.Empty;
        public string Type { get; set; } = string.Empty; // 'issue', 'payment', 'order', 'general'
        public string Priority { get; set; } = string.Empty; // 'low', 'medium', 'high', 'urgent'
        public bool IsRead { get; set; } = false;
        public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
        public DateTime? ReadDate { get; set; }
        
        // Optional references to related entities
        public int? IssueId { get; set; }
        public Issue? Issue { get; set; }
        
        public int? OrderId { get; set; }
        public Order? Order { get; set; }
        
        public int? PaymentId { get; set; }
        public Payment? Payment { get; set; }
    }
} 